flutter/README.md

```diff
@@ -38,7 +42,31 @@ Future<void> main() async {
 }
 ```
 
-Or, if you want to run your app in your own error zone [runZonedGuarded](https://api.flutter.dev/flutter/dart-async/runZonedGuarded.html):
+Starting with Flutter version 3.3, if you want to use the `PlatformDispatcher.onError` callback on your own:
+
+```dart
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+Future<void> main() async {
+  await SentryFlutter.init(
+    (options) {
+      options.dsn = 'https://example@sentry.io/add-your-dsn-here';
+    },
+  );
+
+  PlatformDispatcher.instance.onError = (Object exception, StackTrace stackTrace) async {
+    // Your code here...
+    await Sentry.captureException(exception, stackTrace: stackTrace);
+  };
```

Comment:
Why do we need that? `SentryFlutter` should install the `onError` automatically.

Reply:
Just a sample for users if they want to use the `onError` callback on their own, for whatever reason, they still need to call through to sentry.

Reply:
This is not needed since `PlatformDispatcher.instance.onError` is installed automatically.
the run zoned guard was needed because it was a Flutter limitation.

Reply:
I stumbled upon this PR, while looking for exactly this piece of information.
Indeed, there is a need to allow the developer to implement the PlatformDispatcher on our own, and handle some custom logic. But since it's a single function, not a list of handlers, if we both setup the handler, we might cancel each other out.
So there needs to be a way of passing along the error to Sentry, once the custom logic has been completed in a way. Exactly like the example mentioned here.
Now, the only thing I'm personally uncertain is if this would affect Sentry in any way, of not setting the Handler on it's own.

Reply:
Sentry doesn't override the handler but wraps it. See https://github.com/getsentry/sentry-dart/blob/main/flutter/lib/src/integrations/on_error_integration.dart

It's only overridden if the user sets it after initializing Sentry.

Reply:
Ahhh I see, the 44th line explains a lot.

So you are actually keeping an instance of whatever was setup as the handler, before Sentry integrates it's own. And then whenever the system issues an `onError` call, you can call the handler that was originally setup, and then handle it on your own.

Thanks for the headsup @ueman , really appreciate it! 👍